authorize/evaluator/headers_evaluator_test.go

```diff
@@ -20,14 +20,74 @@ import (
 	"google.golang.org/protobuf/types/known/structpb"
 	"google.golang.org/protobuf/types/known/timestamppb"
 
+	"github.com/pomerium/datasource/pkg/directory"
 	"github.com/pomerium/pomerium/authorize/internal/store"
 	"github.com/pomerium/pomerium/config"
 	"github.com/pomerium/pomerium/pkg/cryptutil"
+	"github.com/pomerium/pomerium/pkg/grpc/databroker"
 	"github.com/pomerium/pomerium/pkg/grpc/session"
 	"github.com/pomerium/pomerium/pkg/grpc/user"
 	"github.com/pomerium/pomerium/pkg/storage"
 )
 
+func BenchmarkHeadersEvaluator(b *testing.B) {
+	ctx := context.Background()
+
+	signingKey, err := cryptutil.NewSigningKey()
+	require.NoError(b, err)
+	encodedSigningKey, err := cryptutil.EncodePrivateKey(signingKey)
+	require.NoError(b, err)
+	privateJWK, err := cryptutil.PrivateJWKFromBytes(encodedSigningKey)
+	require.NoError(b, err)
+	iat := time.Unix(1686870680, 0)
+
+	ctx = storage.WithQuerier(ctx, storage.NewStaticQuerier([]proto.Message{
+		&session.Session{Id: "s1", ImpersonateSessionId: proto.String("s2"), UserId: "u1"},
+		&session.Session{Id: "s2", UserId: "u2", Claims: map[string]*structpb.ListValue{
+			"name": {Values: []*structpb.Value{
+				structpb.NewStringValue("n1"),
+			}},
+		}, IssuedAt: timestamppb.New(iat)},
+		&user.User{Id: "u2", Name: "USER#2"},
+		newDirectoryUserRecord(directory.User{ID: "u2", GroupIDs: []string{"g1", "g2", "g3", "g4"}}),
+		newDirectoryGroupRecord(directory.Group{ID: "g1", Name: "GROUP1"}),
```

Comment:
I think we also populate group emails 
https://github.com/pomerium/pomerium/blob/9f606a9fe92a4f12df793bd35d8679f4a4a5d0c3/authorize/evaluator/opa/policy/headers.rego#L68

Reply:
Strangely that code appears unused. I think we did populate emails at one time but somewhere along the way that support was dropped. Probably on accident.

Reply:
I added some groups to confirm the existing behavior in the JWT test. Emails weren't included.